ratatui-widgets/src/scrollbar.rs

```diff
@@ -136,15 +136,32 @@ pub enum ScrollbarOrientation {
 ///
 /// ```text
 /// ┌───────────────┐
-/// │1. this is a   █
+/// │1. First item  █
+/// │2. Second item ║
+/// └───────────────┘
+/// ```
+///
+/// This example contains also 4 bullets point but each item takes up 2 lines:
+///
+/// - the `content_length` is 8
+/// - the `position` is 0
+/// - the `viewport_content_length` is 4
+///
+/// ```text
+/// ┌───────────────┐
+/// │1. This is a   █
 /// │   single item █
-/// │2. this is a   ║
+/// │2. This is a   ║
 /// │   second item ║
 /// └───────────────┘
 /// ```
 ///
 /// If you don't have multi-line content, you can leave the `viewport_content_length` set to the
 /// default and it'll use the track size as a `viewport_content_length`.
+///
+/// Note that this scrollbar behaves differently to what you might be used to for example
+/// from your web browser in that it scrolls when selecting an item,
+/// regardless of whether all content fits into the current viewport.
```

Comment:
This is a bit ambiguous. When reading this for the first time, I don't understand what you mean by "it scrolls", also there's no selected item in the scrollbar, only a position. While it might be possible to infer this from the context, I think it would be useful to the language align with the names of the properties.

The flexibility of the scrollbar is that it can be used for dealing with either items or lines, but it's the app that chooses how that works. That makes it difficult to document.

Reply:
Yes, that's also not precise enough. I think I have equated it to the selected item and not the viewport position. Maybe providing an example would be helpful?

Reply:
The table example also behave likes this, so we should propably introduce another example.

https://github.com/ratatui/ratatui/blob/main/ratatui/examples/table.rs

Reply:
Same comment as above regarding updating the comments here. I'm looking for suggestions from you (as someone digging into this code and likely the current expert on how it works) to instruct on how best to summarize and communicate what you've learnt. The problem with the current ambiguity is that it states only how something doesn't work by with an ambiguous comparison, without then really clearly stating how it does behave.